### DISCUSSION TITLE
I’m getting an error while trying to get yank highlighting to work

### DISCUSSION THREAD

**Discussion by Folaht** [neovim/neovim#31346] [Q&A] [answered]
The error I recieve

```
Error detected while processing /home/bas/.config/nvim/plugin/options.lua:                                                                                                                      
E5112: Error while creating lua chunk: /home/bas/.config/nvim/plugin/options.lua:24: '=' expected near 'highlight_yank'                                                                         
Press ENTER or type command to continue                                                                                      
```

_~/.config/nvim/plugin/options.lua_
```
...
augroup highlight_yank
    autocmd!
    au TextYankPost * silent! lua vim.highlight.on_yank{higroup="IncSearch", timeout=700}
augroup END
```

**Comment by telemachus** [comment, marked as answer]
I'm not sure I understand how you're trying to do this. `options.lua` should be (?) Lua code, but most of what you show here is Vimscript.

A pure Lua version would look like this.

```lua
local highlight_yank = vim.api.nvim_create_augroup("Whatever-you-want-here", { clear = true })
vim.api.nvim_create_autocmd("TextYankPost", {
    pattern = "*",
    callback = function()
        vim.highlight.on_yank({ higroup = "IncSearch", timeout = 700 })
        -- For anyone in the future who gets here via search, you may want `vim.hl.on_yank`.
        -- See https://github.com/neovim/neovim/pull/30840
        -- vim.hl.on_yank({ higroup = "IncSearch", timeout = 700 })
    end,
    group = highlight_yank,
})
```

**Reply by Folaht** [reply]
Well it was on the Neovim docs on the lua page:
https://neovim.io/doc/user/lua.html#vim.highlight

I however receive an error with this code:
```
Error executing lua callback: /home/bas/.config/nvim/plugin/options.lua:28: attempt to call global 'on_yank' (a nil value)
```

Or when I try `vim.hl.on_yank(..)`
```
Error executing lua callback: /home/bas/.config/nvim/plugin/options.lua:28: attempt to index field 'hl' (a nil value)   
```

[Edit]

`vim.highlight.on_yank(..)` works.

**Reply by telemachus** [reply]
Sorry: I forgot that I had assigned `vim.highlight.on_yank` to a `local on_yank` in my configuration. (I edited my original answer to fix this and to mention the recent change in name from `vim.highlight` to `vim.hl`.)

**Reply by telemachus** [reply]
> Well it was on the Neovim docs on the lua page: https://neovim.io/doc/user/lua.html#vim.highlight

Yes, but they do suggest "in your `init.vim`." That is, they mean for that code to go into a vimscript file rather than a Lua one.

Re the name confusion, `vim.highlight` was [recently renamed to `vim.hl`](https://github.com/neovim/neovim/pull/30840), and the docs online track the master branch. You are probably using a version of neovim that does not yet have that change. See [this comment on a similar question](https://github.com/neovim/neovim/issues/31210#issuecomment-2476511467), for example.

**Comment by clason** [comment]
> Well it was on the Neovim docs on the lua page:
https://neovim.io/doc/user/lua.html#vim.highlight

That page corresponds to the development version (what will be Nvim 0.11), which is going through a number of breaking changes (also relating to this API).

TL;DR: Always use `:help` locally!
